File: common-rule-engine/src/test/files/checks/SwitchWithTooManyCaseCheck/SwitchWithTooManyCaseCheck.go
package main

func foo(tag int) {
   switch tag { // Noncompliant {{Reduce the number of switch cases from 31 to at most 30.}}
// ^^^^^^
	default:
		bar()
	case 0, 1, 2, 3:
		bar()
	case 4, 5, 6, 7:
		bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
	}

}

func foo(tag int) {
  switch tag {
	default:
		bar()
	case 0, 1, 2, 3:
		bar()
	case 4, 5, 6, 7:
		bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  }
}
func bar(s ... string) {
}

File: common-rule-engine/src/test/files/checks/SwitchWithTooManyCaseCheck/SwitchWithTooManyCaseCheck.java
class A{
  void foo() {
    switch (1) { // Noncompliant {{Reduce the number of switch cases from 38 to at most 30.}}
//  ^^^^^^
      case 1:
        System.out.println("");
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      System.out.println("");
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
      case 1:
      case 1:
      case 1:
        break;
      System.out.println("");
      default:
        System.out.println("");
    }

    switch (1) {
      case 1:
      case 1:
      case 1:
        System.out.println("");
      case 1:
      case 1:
      case 1:
        System.out.println("");
    }
  }
}
